pkg/controllers/termination/terminate.go

```diff
@@ -61,6 +61,14 @@ func (t *Terminator) cordon(ctx context.Context, node *v1.Node) error {
 	// 2. Cordon node
 	persisted := node.DeepCopy()
 	node.Spec.Unschedulable = true
+	// Handle nil map
+	if node.Labels == nil {
+		node.Labels = map[string]string{
+			"node.kubernetes.io/exclude-from-external-load-balancers": "karpenter",
```

Comment:
Can use constant [`v1.LoadNodeExcludeLoadBalancers`](https://github.com/kubernetes/api/blob/master/core/v1/well_known_labels.go#L67) instead

Reply:
Nice, didn't know that existed!